test/functional_tests.js

```diff
@@ -89,4 +89,61 @@ describe("basic functional tests", function() {
 
   });
 
+  it("one shield-study-addon telemetry ping for the esper-init event", async() => {
+
+    const foundPings = utils.searchTelemetry([
+      ping => ping.type === "shield-study-addon" && ping.payload.data.attributes.event === "esper-init",
+    ], pings);
+    assert(foundPings.length === 1);
+
+  });
+
+  it("one proper shield-study-addon telemetry ping for the telemetry-payload event", async() => {
+
+    const foundPings = utils.searchTelemetry([
+      ping => ping.type === "shield-study-addon" && ping.payload.data.attributes.event === "telemetry-payload",
+    ], pings);
+    assert(foundPings.length === 1);
+
+    const ping = foundPings[0];
+
+    assert(ping.payload.shield_version === "4.1.0", "expected shield-study-utils version");
+
+    const assertion = value => {
+      return value !== "undefined"
+    };
+
+    const expected = {
+      "completeTelemetrySessionPayload": true,
+      "uptime": true,
+      "total_time": true,
+      "profile_subsession_counter": true,
+      "subsession_start_date": true,
+      "timezone_offset": true,
+      "places_bookmarks_count": true,
+      "places_pages_count": true,
+      "search_counts": true,
+      "scalar_parent_browser_engagement_window_open_event_count": true,
+      "scalar_parent_browser_engagement_total_uri_count": true,
+      "scalar_parent_browser_engagement_navigation_urlbar": true,
+      "scalar_parent_browser_engagement_navigation_contextmenu": true,
+      "scalar_parent_browser_engagement_tab_open_event_count": true,
+      "scalar_parent_browser_engagement_navigation_searchbar": true,
+      "scalar_parent_browser_engagement_navigation_about_newtab": true,
+      "scalar_parent_browser_engagement_unique_domains_count": true,
+      "scalar_parent_browser_engagement_max_concurrent_window_count": true,
+      "scalar_parent_browser_engagement_max_concurrent_tab_count": true,
+      "scalar_parent_browser_engagement_unfiltered_uri_count": true,
+    };
+
+    const actual = {};
+
+    for (const attribute in expected) {
```

Comment:
Simpler might be:  "deepEqual(expectedKeys.sorted(), Object.keys(thePing).sorted() )" .  This also is clear.